packages/shared/lib/utils/utils.ts

```diff
@@ -366,6 +366,22 @@ export function getConnectionMetadataFromTokenResponse(params: any, provider: Pr
     return combinedArr.length > 0 ? (Object.fromEntries(combinedArr) as Record<string, any>) : {};
 }
 
+/**
+ * A helper function to extract the additional connection metadata returned from the Provider in the token response.
+ * It can parse booleans or strings only
+ */
+export function getConnectionMetadataFromTokenResponse(params: any, provider: Provider): Record<string, any> {
+    return getConnectionMetadata(params, provider, 'token_response_metadata');
+}
+
+/**
+ * A helper function to extract the additional connection metadata returned from the Provider in the webhook response.
+ * It can parse booleans or strings only
+ */
+export function getConnectionMetadataFromWebhookResponse(params: any, provider: Provider): Record<string, any> {
+    return getConnectionMetadata(params, provider, 'webhook_response_metadata');
+}
+
```

Comment:
are those function necessary? Can `getConnectionMetadata` be called with the extra parameters instead?